server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -130,7 +130,7 @@ public static HighlightBuilder highlight() {
 
     private QueryBuilder postQueryBuilder;
 
-    private KnnSearchBuilder knnSearch;
+    private List<KnnSearchBuilder> knnSearch = new ArrayList<>();
```

Comment:
could we make this final? instead of taking a list we could take KnnSearchBuilders and add to the initial list internally

Reply:
@javanna none of the other list field values behave this way. I would rather have consistency so as not to create surprises.